include/picongpu/CMakeLists.txt

```diff
@@ -277,6 +277,15 @@ if(openPMD_FOUND)
     if(openPMD_HAVE_ADIOS2 OR openPMD_HAVE_HDF5)
         message(STATUS "Found openPMD: ${openPMD_DIR}")
         add_definitions(-DENABLE_OPENPMD=1)
+
+        # non of these should appear in cmake-gui, so make them internal
+        set(JSON_BuildTests OFF CACHE INTERNAL "")
+        set(JSON_MultipleHeaders OFF CACHE INTERNAL "")
+        set(JSON_ImplicitConversions OFF CACHE INTERNAL "")
+        set(JSON_Install OFF CACHE INTERNAL "")  # only used PRIVATE
+        add_subdirectory(
```

Comment:
Could you please add a witch to use an external version wi `find_package` or the internal version and then with `add_subdirectory`

example:
https://github.com/ComputationalRadiationPhysics/picongpu/blob/fa864412e591870d04641d48d5ff58289112f35c/thirdParty/cupla/cuplaConfig.cmake#L112-L122

Reply:
Sure, I can add that

Reply:
Done. I suppose, the CI containers should be updated with a build run that pulls the most recent nlohmann_json?

Reply:
Possible but not required. This can be shipped in a follow up PR.